js/source/tile.js

```diff
@@ -138,7 +138,12 @@ Tile.prototype = {
     },
 
     getElementGroups: function(layer, shaderName) {
-        return this.elementGroups && this.elementGroups[layer.ref || layer.id] && this.elementGroups[layer.ref || layer.id][shaderName];
+        if (!this.buffers || !Object.keys(this.buffers)) return null;
```

Comment:
I understand the problem a little better now. `Tile#buffers` is null because of https://github.com/mapbox/mapbox-gl-js/blob/null-buffers-1983/js/source/tile.js#L57 and https://github.com/mapbox/mapbox-gl-js/blob/null-buffers-1983/js/source/tile.js#L109. I'm going to remove `!Object.keys(this.buffers).length` entirely.